src/units/abilities.cpp

```diff
@@ -2324,39 +2324,32 @@ bool attack_type::special_tooltip_active(const config& special, const std::strin
 {
 	//log_scope("special_tooltip_active");
 
+	//here 'active_on' and checking of opponent weapon shouldn't implemented
+	//because other_attack_ don't exist in sidebar display.
+	//'apply_to' and some filters like [filter_student] are checked for know if
+	//special must be displayed in sidebar.
+
+	//only special who affect self are valid here.
 	bool whom_is_self = special_affects_self(special, is_attacker_);
-	if(!whom_is_self)
+	if(!whom_is_self) {
 		return false;
-
-	const unit_map& units = get_unit_map();
-
-	unit_const_ptr self = self_ ;
-	unit_const_ptr other = other_;
-	//TODO: why is this needed?
-	if(self == nullptr) {
```

Comment:
Are you removing this knowing why it is not needed or simply hoping it is not?

The `other` use seems ineffective since this function intentionally ignores other units as I understand it though if `self_` is invalid but there is a unit on `self_loc_` then this seems like the right thing to do.

Reply:
i _think_ that no such case exists exists, iirc the only case were self_ is null but self_loc_ is a valid location is that code that checks "how would a unit of a given type perform" in a ai recruitment path, in which case using the unit at the given location would be wrong.

But usually we want the commit message that removed the code to have a little explanation why it does that.


partly related: i think it would be worth good to remove that codepath completely and just make the ai create a dummy unit.

Reply:
> @gfgtdf note that this is code that duplicated your TODO comment from special_active (now special_active_impl).

yes, i'm pretty sure thatthe reasonong for writing that comment was whats i just said,  and that i just wanted to check it again.

What i'm really wondering though is why that code duplication was needed.

Reply:
i will remove this code because self_ exist ever for sidebar check